From 7252faadcbe7fb17f2eaa3381bb2b795f2e542e2 Mon Sep 17 00:00:00 2001
From: "Brett T. Warden" <brett.t.warden@intel.com>
Date: Thu, 1 Nov 2018 17:17:56 -0700
Subject: [PATCH 2/2] Set better key default prefs

---
 g10/gpg.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/g10/gpg.c b/g10/gpg.c
index e18eefe..bd3c399 100644
--- a/g10/gpg.c
+++ b/g10/gpg.c
@@ -2397,6 +2397,9 @@ main (int argc, char **argv)
     opt.emit_version = 0;
     opt.weak_digests = NULL;
 
+    /* Set more secure default ciphers and hashes for key generation */
+    opt.def_preference_list = "SHA512 SHA384 SHA256 AES256";
+
     /* Check whether we have a config file on the command line.  */
     orig_argc = argc;
     orig_argv = argv;
-- 
2.19.1

