From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Brett T. Warden" <brett.t.warden@intel.com>
Date: Thu, 1 Nov 2018 17:17:56 -0700
Subject: [PATCH] Set better key default prefs

---
 g10/gpg.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/g10/gpg.c b/g10/gpg.c
index 431c516..00048c7 100644
--- a/g10/gpg.c
+++ b/g10/gpg.c
@@ -2432,6 +2432,9 @@ main (int argc, char **argv)
     opt.emit_version = 0;
     opt.weak_digests = NULL;
 
+    /* Set more secure default ciphers and hashes for key generation */
+    opt.def_preference_list = "SHA512 SHA384 SHA256 AES256";
+
     /* Check special options given on the command line.  */
     orig_argc = argc;
     orig_argv = argv;
